holmes/core/tool_calling_llm.py

```diff
@@ -126,9 +123,12 @@ def get_maximum_output_token(self) -> int:
         except Exception as e:
             logging.warning(f"Couldn't find model's name {model_name} in litellm's model list, fallback to 4096 tokens for max_output_tokens")
             return 4096
-
-    def call(self, system_prompt, user_prompt, post_process_prompt: Optional[str] = None, response_format: dict = None) -> LLMResult:
-        messages = [
+    
+    def call(self, system_prompt: Optional[str] = None, user_prompt: Optional[str] = None, post_process_prompt: Optional[str] = None, response_format: dict = None,
```

Comment:
I think this is confusing 
system_prompt, user_prompt, and messages cannot be all optional

I think it would be clearer if you separate it to 2 functions, which hold the common code in a third function

looks like you have `messages_call` and `prompt_call`